test/support/test_config.h

```diff
@@ -326,4 +326,28 @@
 // Drop view throws exceptions in libstdc++ 10
 #define _PSTL_LIBSTDCXX_XPU_DROP_VIEW_BROKEN (_GLIBCXX_RELEASE == 10)
 
+// std::ranges::view concept in libstdc++ 10 is broken, and in GCC 11 is broken prior to GCC 11.4 (i.e., __GLIBCXX__ < 20230714)
+#if defined(_GLIBCXX_RELEASE) && defined(__GLIBCXX__)
+#    define TEST_GCC10_STD_RANGES_VIEW_CONCEPTS_BROKEN                                                                 \
+        ((_GLIBCXX_RELEASE == 10) || (_GLIBCXX_RELEASE == 11 && __GLIBCXX__ < 20230714))
+#else
+#    define TEST_GCC10_STD_RANGES_VIEW_CONCEPTS_BROKEN 0
+#endif
+
+// std::ranges::views::all in libstdc++ 10 is broken prior to GCC 12.1
+#if defined(_GLIBCXX_RELEASE) && defined(__GLIBCXX__)
+#    define TEST_GCC10_STD_RANGES_VIEW_ALL_BROKEN                                                                      \
+        ((_GLIBCXX_RELEASE == 10) || (_GLIBCXX_RELEASE == 11 && __GLIBCXX__ < 20220507))

```

Comment:
It seems it is not a bug in libstdc++. My understanding is that [p2415](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2415r2.html) introduced `owning_view`, which allowed constructing `all_view` from an r-value. This paper was implemented in libstdc++ 12.1.

The case you guard with this macro creates all_view from an r-value, which should become this newly allowed owning_view:
```
auto all_view = std::ranges::views::all(MinimalisticRangeForIntVec(vec.begin(), vec.end()));
```

My suggestion is to change the macro name to denote it:
`TEST_GCC10_STD_RANGES_VIEW_ALL_BROKEN` -> `TEST_HAS_OWNING_VIEW`.

Reply:
Also, what do you think about changing the test case to construct all_view from an l-value to avoid adding this macro?